tests/unit/s2n_ktls_test.c

```diff
@@ -327,6 +333,25 @@ int main(int argc, char **argv)
             server_conn->managed_recv_io = true;
             EXPECT_SUCCESS(s2n_connection_ktls_enable_recv(server_conn));
         };
+
+        /* Fail if renegotiation potentially supported */
+        {
+            DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+
+            DEFER_CLEANUP(struct s2n_connection *client = s2n_connection_new(S2N_CLIENT),
+                    s2n_connection_ptr_free);
+            EXPECT_OK(s2n_test_configure_connection_for_ktls(client));
+            EXPECT_SUCCESS(s2n_connection_set_config(client, config));
+
+            DEFER_CLEANUP(struct s2n_connection *server = s2n_connection_new(S2N_SERVER),
+                    s2n_connection_ptr_free);
+            EXPECT_OK(s2n_test_configure_connection_for_ktls(server));
+            EXPECT_SUCCESS(s2n_connection_set_config(server, config));
+
+            EXPECT_SUCCESS(s2n_config_set_renegotiate_request_cb(config, s2n_test_reneg_cb, NULL));
+            EXPECT_FAILURE_WITH_ERRNO(s2n_connection_ktls_enable_recv(client), S2N_ERR_KTLS_RENEG);
```

Comment:
Why do we want this to only error for the client?

Reply:
We only support server-initiated renegotiation, meaning that only the client receives HelloRequest messages, triggers the callback, and renegotiates. If the callback is set on a server, it's ignored. That's covered by this line: https://github.com/aws/s2n-tls/blob/bdbcaab1e84b502cc54a5479bc3b9a9de26000df/tls/s2n_ktls.c#L81 servers can't receive hello requests.